src/internal_events/common.rs

```diff
@@ -132,3 +132,72 @@ impl InternalEvent for ComponentEventsDropped {
         );
     }
 }
+
+#[derive(Debug)]
+pub struct SinkRequestBuildError<E> {
+    pub name: &'static str,
+    pub error: E,
+}
+
+impl<E: std::fmt::Display> InternalEvent for SinkRequestBuildError<E> {
+    fn emit(self) {
+        error!(
+            message = format!("Failed to build request for {}", self.name),
+            error = %self.error,
+            error_type = error_type::ENCODER_FAILED,
+            stage = error_stage::PROCESSING,
+            rate_limit_secs = 10,
+        );
+        counter!(
+            "component_errors_total", 1,
+            "error_type" => error_type::ENCODER_FAILED,
+            "stage" => error_stage::PROCESSING,
+        );
+    }
+}
+
+#[derive(Debug)]
+pub struct SinkRetryReasonError<'a> {
+    pub message: &'static str,
+    pub reason: Cow<'a, str>,
+}
+
+impl InternalEvent for SinkRetryReasonError<'_> {
+    fn emit(self) {
+        error!(
+            message = %self.message,
+            reason = %self.reason,
+            error_type = error_type::REQUEST_FAILED,
+            stage = error_stage::SENDING,
+            rate_limit_secs = 10,
+        );
+        counter!(
+            "component_errors_total", 1,
+            "error_type" => error_type::REQUEST_FAILED,
+            "stage" => error_stage::SENDING,
+        );
+    }
+}
+
+#[derive(Debug)]
+pub struct SinkRetryError<E> {
+    pub message: &'static str,
+    pub error: E,
+}
+
+impl<E: std::fmt::Display> InternalEvent for SinkRetryError<E> {
+    fn emit(self) {
+        error!(
+            message = %self.message,
+            error = %self.error,
+            error_type = error_type::ENCODER_FAILED,
```

Comment:
Encoding failed for a "RetryError"? Seems wrong.

Reply:
I couldn't find a better one. Is it ok to add to that `error_type` mod ?

Reply:
Addressed in https://github.com/vectordotdev/vector/pull/14225/commits/de926d09da37f5ec27f2f52ddc4e163e8336944a